hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -528,7 +505,7 @@ private void initializeFileGroups(HoodieTableMetaClient dataMetaClient, Metadata
   private <T> void processAndCommit(String instantTime, ConvertMetadataFunction convertMetadataFunction, boolean canTriggerTableService) {
     if (enabled && metadata != null) {
       List<HoodieRecord> records = convertMetadataFunction.convertMetadata();
-      commit(records, MetadataPartitionType.FILES.partitionPath(), instantTime, canTriggerTableService);
+      commit(engineContext.parallelize(records, 1), MetadataPartitionType.FILES.partitionPath(), instantTime, canTriggerTableService);
```

Comment:
not sure if we need to parallelize with records.size(). for regular delta commits, 1 should be good enough. Bootstrap code path is different.

Reply:
Yep. thats seems right.